### PR TITLE
storage: Fix scan_lock doesn't return lock's TTL

### PR DESCRIPTION
**Pull request by MyonKeminta**
Signed-off-by: MyonKeminta <MyonKeminta@users.noreply.github.com>

###  What have you changed?

TiKV's scan_lock didn't pass the locks' TTLs. So when GC meets lock, TiDB will get 0 as the TTL, and all locks will be regarded as expired and cleaned up. Unfinished transactions will be aborted.

This PR fixes this problem.

###  What is the type of the changes?

Pick one of the following and delete the others:

- Bugfix (a change which fixes an issue)

###  How is the PR tested?

- Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No

###  Does this PR affect `tidb-ansible`?

No

### CODE REVIEW

**Comment by sticnarf** [comment]
To my understanding, all locks should be resolved before safepoint regardless TTL?

**Comment by MyonKeminta** [comment]
@sticnarf I think it's not very important, since this situation is very rare.

File: src/storage/txn/process.rs [hunk 1]
```diff
@@ -416,6 +416,7 @@ fn process_read_impl<E: Engine>(
                 lock_info.set_primary_lock(lock.primary);
                 lock_info.set_lock_version(lock.ts);
                 lock_info.set_key(key.into_raw()?);
+                lock_info.set_lock_ttl(lock.ttl);
```
Comment (overvenus) [inline src/storage/txn/process.rs]: Should we set `txn_size`?

**Comment by MyonKeminta** [comment]
@overvenus @sticnarf PTAL again

**Pull request merged**